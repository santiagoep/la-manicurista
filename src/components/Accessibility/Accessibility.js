import React, { useState } from "react";
import { useTranslation } from "react-i18next";

import useConfig from "@hooks/useConfig";
import { fontSizes } from "@utils/constants";
import { StyledContainer, StyledItem } from "./Accessibility.styled";

const Accessibility = () => {
  const { t } = useTranslation();
  const [currentFontSize, setCurrentFontSize] = useState(1);
  const [
    { fontSize, theme, language },
    { setFontSize, toggleLanguage, toggleTheme },
  ] = useConfig();

  const toggleFontSize = () => {
    const fontSizesArr = Object.values(fontSizes);
    if (fontSizesArr[currentFontSize + 1]) {
      setFontSize(fontSizesArr[currentFontSize + 1]);
      setCurrentFontSize(currentFontSize + 1);
    } else {
      setFontSize(fontSizesArr[0]);
      setCurrentFontSize(0);
    }
  };

  return (
    <StyledContainer>
      <StyledItem className="accessibility__font-size" onClick={toggleFontSize}>
        {`${t("fontSize")}:${fontSize}`}
      </StyledItem>
      <StyledItem className="accessibility__theme" onClick={toggleTheme}>
        {`${t("theme")}:${theme}`}
      </StyledItem>
      <StyledItem className="accessibility__language" onClick={toggleLanguage}>
        {`${t("language")}:${language}`}
      </StyledItem>
    </StyledContainer>
  );
};

export default Accessibility;
