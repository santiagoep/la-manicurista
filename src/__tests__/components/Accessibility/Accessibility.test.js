import React from "react";

import render from "../../render";
import Accessibility from "@components/Accessibility/Accessibility";

describe("<Accessibility />", () => {
  it("should render snapshot", () => {
    const { renderer } = render({ theme: true, redux: true });
    const component = renderer(<Accessibility />);
    expect(component).toMatchSnapshot();
  });
  it("should render click on fontSize", () => {
    const { mount } = render({ theme: true, redux: true });
    const component = mount(<Accessibility />);
    component.find(".accessibility__font-size").first().simulate("click");
    expect(component).toMatchSnapshot();
  });
  it("should render click on theme", () => {
    const { mount } = render({ theme: true, redux: true });
    const component = mount(<Accessibility />);
    component.find(".accessibility__theme").first().simulate("click");
    expect(component).toMatchSnapshot();
  });
  it("should render click on language", () => {
    const { mount } = render({ theme: true, redux: true });
    const component = mount(<Accessibility />);
    component.find(".accessibility__language").first().simulate("click");
    expect(component).toMatchSnapshot();
  });
});
