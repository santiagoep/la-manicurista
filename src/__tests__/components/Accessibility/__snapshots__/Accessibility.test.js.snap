// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accessibility /> should render click on fontSize 1`] = `
<WrapperComponent
  config={
    Object {
      "redux": true,
      "theme": true,
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "breakpoints": Object {
            "lg": "@media only screen and (min-width: 75em)",
            "md": "@media only screen and (min-width: 64em)",
            "sm": "@media only screen and (min-width: 48em)",
            "xs": "@media only screen and (min-width: 0em)",
          },
          "colors": Object {
            "bg": Object {
              "playerHeader": "rgba(236, 240, 241, 0.9)",
              "primary": "#FF0177",
              "quaternary": "#919191",
              "quinary": "#4E8ECA",
              "secondary": "#0D3B66",
              "tertiary": "#FFF",
            },
            "borders": Object {
              "active": "#FF0177",
              "primary": "#FF0177",
              "secondary": "#4E8ECA",
            },
            "font": Object {
              "active": "#FF0177",
              "danger": "#FF2624",
              "primary": "#FF0177",
              "quaternary": "#FFF",
              "quinary": "#333",
              "secondary": "#0D3B66",
              "tertiary": "#919191",
            },
          },
          "font": Object {
            "size": Object {
              "caption": "0.75rem",
              "h1": "3.75rem",
              "h2": "3rem",
              "h3": "2.5rem",
              "h4": "2.25rem",
              "h5": "1.625rem",
              "h6": "1.5rem",
              "help": "0.625em",
              "lg": "18px",
              "md": "16px",
              "paragraph1": "1.125rem",
              "paragraph2": "1rem",
              "paragraph3": "0.875rem",
              "sm": "14px",
              "small": "0.5rem",
              "subtitle1": "1.1375rem",
              "subtitle2": "1.25rem",
              "subtitle3": "1.375rem",
            },
          },
          "name": "LIGHT",
          "utils": Object {
            "spacing": [Function],
          },
        }
      }
    >
      <Accessibility>
        <Accessibilitystyled__StyledContainer>
          <span
            className="Accessibilitystyled__StyledContainer-gi4yzg-0 dWjzqH"
          >
            <Accessibilitystyled__StyledItem
              className="accessibility__font-size"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__font-size"
                onClick={[Function]}
              >
                Font size:lg
              </div>
            </Accessibilitystyled__StyledItem>
            <Accessibilitystyled__StyledItem
              className="accessibility__theme"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__theme"
                onClick={[Function]}
              >
                Tema:LIGHT
              </div>
            </Accessibilitystyled__StyledItem>
            <Accessibilitystyled__StyledItem
              className="accessibility__language"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__language"
                onClick={[Function]}
              >
                Idioma:es
              </div>
            </Accessibilitystyled__StyledItem>
          </span>
        </Accessibilitystyled__StyledContainer>
      </Accessibility>
    </Component>
  </Provider>
</WrapperComponent>
`;

exports[`<Accessibility /> should render click on language 1`] = `
<WrapperComponent
  config={
    Object {
      "redux": true,
      "theme": true,
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "breakpoints": Object {
            "lg": "@media only screen and (min-width: 75em)",
            "md": "@media only screen and (min-width: 64em)",
            "sm": "@media only screen and (min-width: 48em)",
            "xs": "@media only screen and (min-width: 0em)",
          },
          "colors": Object {
            "bg": Object {
              "playerHeader": "rgba(236, 240, 241, 0.9)",
              "primary": "#FF0177",
              "quaternary": "#919191",
              "quinary": "#4E8ECA",
              "secondary": "#0D3B66",
              "tertiary": "#FFF",
            },
            "borders": Object {
              "active": "#FF0177",
              "primary": "#FF0177",
              "secondary": "#4E8ECA",
            },
            "font": Object {
              "active": "#FF0177",
              "danger": "#FF2624",
              "primary": "#FF0177",
              "quaternary": "#FFF",
              "quinary": "#333",
              "secondary": "#0D3B66",
              "tertiary": "#919191",
            },
          },
          "font": Object {
            "size": Object {
              "caption": "0.75rem",
              "h1": "3.75rem",
              "h2": "3rem",
              "h3": "2.5rem",
              "h4": "2.25rem",
              "h5": "1.625rem",
              "h6": "1.5rem",
              "help": "0.625em",
              "lg": "18px",
              "md": "16px",
              "paragraph1": "1.125rem",
              "paragraph2": "1rem",
              "paragraph3": "0.875rem",
              "sm": "14px",
              "small": "0.5rem",
              "subtitle1": "1.1375rem",
              "subtitle2": "1.25rem",
              "subtitle3": "1.375rem",
            },
          },
          "name": "LIGHT",
          "utils": Object {
            "spacing": [Function],
          },
        }
      }
    >
      <Accessibility>
        <Accessibilitystyled__StyledContainer>
          <span
            className="Accessibilitystyled__StyledContainer-gi4yzg-0 dWjzqH"
          >
            <Accessibilitystyled__StyledItem
              className="accessibility__font-size"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__font-size"
                onClick={[Function]}
              >
                Font size:md
              </div>
            </Accessibilitystyled__StyledItem>
            <Accessibilitystyled__StyledItem
              className="accessibility__theme"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__theme"
                onClick={[Function]}
              >
                Tema:LIGHT
              </div>
            </Accessibilitystyled__StyledItem>
            <Accessibilitystyled__StyledItem
              className="accessibility__language"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__language"
                onClick={[Function]}
              >
                Idioma:en
              </div>
            </Accessibilitystyled__StyledItem>
          </span>
        </Accessibilitystyled__StyledContainer>
      </Accessibility>
    </Component>
  </Provider>
</WrapperComponent>
`;

exports[`<Accessibility /> should render click on theme 1`] = `
<WrapperComponent
  config={
    Object {
      "redux": true,
      "theme": true,
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "breakpoints": Object {
            "lg": "@media only screen and (min-width: 75em)",
            "md": "@media only screen and (min-width: 64em)",
            "sm": "@media only screen and (min-width: 48em)",
            "xs": "@media only screen and (min-width: 0em)",
          },
          "colors": Object {
            "bg": Object {
              "playerHeader": "rgba(236, 240, 241, 0.9)",
              "primary": "#FF0177",
              "quaternary": "#919191",
              "quinary": "#4E8ECA",
              "secondary": "#0D3B66",
              "tertiary": "#FFF",
            },
            "borders": Object {
              "active": "#FF0177",
              "primary": "#FF0177",
              "secondary": "#4E8ECA",
            },
            "font": Object {
              "active": "#FF0177",
              "danger": "#FF2624",
              "primary": "#FF0177",
              "quaternary": "#FFF",
              "quinary": "#333",
              "secondary": "#0D3B66",
              "tertiary": "#919191",
            },
          },
          "font": Object {
            "size": Object {
              "caption": "0.75rem",
              "h1": "3.75rem",
              "h2": "3rem",
              "h3": "2.5rem",
              "h4": "2.25rem",
              "h5": "1.625rem",
              "h6": "1.5rem",
              "help": "0.625em",
              "lg": "18px",
              "md": "16px",
              "paragraph1": "1.125rem",
              "paragraph2": "1rem",
              "paragraph3": "0.875rem",
              "sm": "14px",
              "small": "0.5rem",
              "subtitle1": "1.1375rem",
              "subtitle2": "1.25rem",
              "subtitle3": "1.375rem",
            },
          },
          "name": "LIGHT",
          "utils": Object {
            "spacing": [Function],
          },
        }
      }
    >
      <Accessibility>
        <Accessibilitystyled__StyledContainer>
          <span
            className="Accessibilitystyled__StyledContainer-gi4yzg-0 dWjzqH"
          >
            <Accessibilitystyled__StyledItem
              className="accessibility__font-size"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__font-size"
                onClick={[Function]}
              >
                Font size:md
              </div>
            </Accessibilitystyled__StyledItem>
            <Accessibilitystyled__StyledItem
              className="accessibility__theme"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__theme"
                onClick={[Function]}
              >
                Tema:DARK
              </div>
            </Accessibilitystyled__StyledItem>
            <Accessibilitystyled__StyledItem
              className="accessibility__language"
              onClick={[Function]}
            >
              <div
                className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__language"
                onClick={[Function]}
              >
                Idioma:es
              </div>
            </Accessibilitystyled__StyledItem>
          </span>
        </Accessibilitystyled__StyledContainer>
      </Accessibility>
    </Component>
  </Provider>
</WrapperComponent>
`;

exports[`<Accessibility /> should render snapshot 1`] = `
<span
  className="Accessibilitystyled__StyledContainer-gi4yzg-0 dWjzqH"
>
  <div
    className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__font-size"
    onClick={[Function]}
  >
    Font size:md
  </div>
  <div
    className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__theme"
    onClick={[Function]}
  >
    Tema:LIGHT
  </div>
  <div
    className="Accessibilitystyled__StyledItem-gi4yzg-1 aetAM accessibility__language"
    onClick={[Function]}
  >
    Idioma:es
  </div>
</span>
`;
